charts/velero/templates/_helpers.tpl

```diff
@@ -89,7 +89,7 @@ Create the backup storage location provider
 */}}
 {{- define "velero.backupStorageLocation.provider" -}}
 {{- with .Values.configuration -}}
-{{ default .backupStorageLocation.provider .provider }}
+{{ default .provider .backupStorageLocation.provider }}
```

Comment:
In the line below, if there's a value for `provider`, it will make that the default value for the `backupStorageLocation.provider` anyway. Unless I'm reading this wrong. If it's right, would this be necessary?

https://github.com/vmware-tanzu/helm-charts/blob/master/charts/velero/templates/_helpers.tpl#L92

Reply:
Currently, if `configuration.provider` is set (and it has to be to enable deployment) it overrides `configuration.backupStorageLocation.provider`. Why? Because the first value is actually the default and the second one if given: [doc](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function)

Reply:
I appreciate the explanation! My understanding was the exact opposite, so glad to have this corrected. Thanks for the link to the relevant docs, I'll revisit that too.